docs/deployment-guide.md

```diff
@@ -0,0 +1,29 @@
+# Deploying Container Object Storage Interface (COSI) On Kubernetes
+
+This document describes steps for Kubernetes administrators to setup Container Object Storage Interface (COSI) onto a Kubernetes cluster.
+
+## Overview
+
+There are four components that need to be deployed in Kubernetes to setup COSI.
+
+- CustomResourceDefinitions (CRDs)
+- Controller
+- Provisioner Sidecar
+- Node Adapter
+
+### CustomResourceDefinitions
+
+COSI acts on following [custom resource definitions (CRDs)](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/):
+
+- [BucketClass](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketclass), which is an immutable, cluster-scoped, custom resource to provide admins control over the handling of bucket provisioning.
+- [BucketRequest](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketrequest), which is a user facing, namespaced custom resource requesting provisioning of a new bucket, or requesting access to an existing bucket.
+- [Bucket](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucket), which is a cluster-scoped custom resource representing the abstraction of a single backend bucket.
+- [BucketAccessRequest](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccessrequest), which is a user namespaced custom resource representing an object store user and an access policy defining the user’s relation to that storage.
+- [BucketAccessClass](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccessclass), which is an immutable, cluster-scoped, custom resource providing a way for admins to specify policies that may be used to access buckets.
+- [BucketAccess](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support#bucketaccess), which is a cluster-scoped administrative custom resource which encapsulates fields from the `BucketAccessRequest (BAR)` and the `BucketAccessClass (BAC)`.
+
+All [COSI custom resource definitions](../crds) can be installed using the [kustomization](https://kustomize.io/) [file](../kustomization.yaml) and `kubectl` with following command:
+
+```sh
+kubectl create -k github.com/kubernetes-sigs/container-object-storage-api
```

Comment:
There is a typo in this one, the word "interface" is missing. The correct one is:
```
kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
```